# Draft project paper

Наша мета — симулювати роботу команди інвесторів, завданням якої є управління портфелем фондів на суму від $1000 до $1 млн на початку 2023 року. Основна ціль полягає в тому, щоб дослідити, порівняти та запропонувати кілька різних інвестиційних альтернатив.

# Інструменти в які ми інвестуємо:

## MSFT (Microsoft Corporation)
Microsoft — це світовий лідер у галузі програмного забезпечення, послуг хмарних обчислень і рішень з цифрової безпеки. Компанія відома своєю операційною системою Windows, офісним пакетом Microsoft Office, хмарним сервісом Azure, а також продуктами у сфері штучного інтелекту та ігрової індустрії, включаючи Xbox. Інвестування в Microsoft нами розглядаэться як вкладення в компанію, що займає стабільні позиції на ринку та має довгостроковий потенціал зростання.

## VTI (Vanguard Total Stock Market ETF)
VTI — це ETF (фонд, що торгується на біржі) від Vanguard, який відтворює індекс усього американського фондового ринку. Він включає акції компаній різного розміру з різних секторів економіки, забезпечуючи диверсифікацію та зниження ризику для інвестора. Це популярний вибір для тих, хто хоче мати інвестиції у великому спектрі американських компаній.

## BTC (Bitcoin)
Bitcoin — перша і найпопулярніша криптовалюта у світі, яка пропонує децентралізовану систему електронних грошей, що не контролюється жодним урядом чи фінансовою інституцією. Інвестування в Bitcoin - це вкладенням в актив з високим потенціалом зростання, хоча й з високим рівнем волатильності та ризику.

## KO (The Coca-Cola Company)
The Coca-Cola Company — одна з найбільших у світі компаній, що займається виробництвом безалкогольних напоїв. Вона відома своїм флагманським продуктом Coca-Cola, а також широким асортиментом інших напоїв. Інвестування в Coca-Cola розглядаэться нами як вкладення в компанію з міцними брендами, стабільним доходом та дивідендною політикою.

## TLT (iShares 20+ Year Treasury Bond ETF)
TLT — це ETF, який інвестує в довгострокові облігації скарбниці США. Цей фонд надає інвесторам можливість отримати дохід у вигляді купонних платежів з облігацій з високим рівнем кредитної надійності. Інвестування в TLT розглядається як вкладення в безпечні активи, особливо під час періодів ринкової невизначеності та волатильності.

Наш вибір охоплює широкий спектр активів — від акцій та ETF, які представляють різні сектори економіки, до криптовалюти та державних облігацій. Це дозволяє диверсифікувати портфель та зменшити ризики, пов’язані з інвестуванням.

## Крок перший: Аналіз ринку
Завдяки платформі Yahoo Invest Шукаємо потрібні нам дані про акції компаній: Ціна акцій на момент 01.01.2023(День коли ми вкладаємо кошти) та на момент 31.12.2023(День коли ми фіксуємо профіт), а також дивіденди.


| Instrument    | Opening price |Closing price| Dividends|
| --------      | -------           |----------|-----------|
| MSFT          |   243.08          |376.04 | 0.71%|
| VTI           |   192.38	        |237.22	 |1.42%|
| BTC           | 16,547.91	        |42,265.19|Nan|
| KO            | 63.56	            |58.27| 3,23%|
| TLT           | 102.20            |91.31|	4.07%|

## Крок перший: Рахуємо кількість акцій/юнітів

Ми вирішили розділити всі кошти порівну між всіма компаніями і отримали по $6000 на кожну

In [3]:
import pandas as pd

# Read the data from the CSV file using pandas
filename = 'TVM.csv'
data = pd.read_csv(filename)

# Convert the 'investment' column to integers
data['investment'] = data['investment'].astype(int)
data['opening_price'] = data['opening_price'].astype(float)
# Calculate the number of shares
data['shares'] = data['investment'] / data['opening_price']

# Write the updated data back to the CSV file
data.to_csv(filename, index=False)
print("Data has been updated and saved to the file:", filename)
print(data[['instrument', 'shares']])



Data has been updated and saved to the file: TVM.csv
   instrument       shares
0       MSFT1    24.683232
1        VTI1    31.188273
2        BTC1     0.362584
3         KO1    94.398993
4        TLT1    58.708415
5       MSFT2   329.109758
6        VTI2   415.843643
7        BTC2     4.834447
8         KO2  1258.653241
9        TLT2   782.778865
10      MSFT3   822.774395
11       VTI3  1039.609107
12       BTC3    12.086118
13        KO3  3146.633103
14       TLT3  1956.947162


In [19]:
import pandas as pd
import numpy_financial as npf

# Read the data from the CSV file using pandas
filename = 'TVM.csv'
data = pd.read_csv(filename)

# Calculate the capital gains for each instrument
data['capital_gains'] = (data['shares'] * data['closing_price']) - (data['shares']*data['opening_price'])

# Calculate the dividend income for each instrument
data['dividend_income'] = data['investment'] * data['dividend_percent']

# Calculate the total end values for each instrument
data['total_end_values'] = data['investment'] + data['capital_gains'] + data['dividend_income']

# Calculate the profit for each instrument
data['profit'] = data['total_end_values'] - data['investment']

# Calculate the profit in %
data['profit_percent'] = (data['profit'] / data['investment']) * 100

# # Calculate the IRR
cash_flows = data['total_end_values'].tolist()
cash_flows.insert(0, -data['investment'].sum())
irr = npf.irr(cash_flows)

# # Write the updated data back to the CSV file
data.to_csv(filename, index=False)
print("Data has been updated and saved to the file:", filename)
print(data[['instrument', 'profit', 'profit_percent', 'total_end_values', 'capital_gains', 'dividend_income']])
print("IRR:", irr)



Data has been updated and saved to the file: TVM.csv
  instrument       profit  profit_percent  total_end_values  capital_gains  \
0       MSFT  3324.482508       55.408042       9324.482508    3281.882508   
1        VTI  1483.682171       24.728036       7483.682171    1398.482171   
2        BTC  9324.662752      155.411046      15324.662752    9324.662752   
3         KO  -305.570673       -5.092845       5694.429327    -499.370673   
4        TLT  -395.134638       -6.585577       5604.865362    -639.334638   

   dividend_income  
0             42.6  
1             85.2  
2              0.0  
3            193.8  
4            244.2  
IRR: 0.14828831990348945


### 


### So as we can see from our calculations
|Instrument| Capital gains|Dividend gains|Total End Values|Profit|IRR|
|----------|--------------|-----------|------------|--------|--------
  |'MSFT'| 3281.88|70.59|9352.48|3352.48|
  |'VTI'| 1398.48|81.93|7480.41|1480.41|
  |'BTC'| 9324.66|0|15324.66|9324.66|
  |'KO'|-437.067|183.13|5746.06|-253.93|
  |'TLT'| -194.91|16.97|5822.05|-177.95|
